iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -65,22 +71,24 @@ internal sealed class HttpClientHelper : IDisposable
             X509Certificate2 clientCert = _connectionCredentials.ClientCertificate;
             if (clientCert != null)
             {
-                _httpClientHandler.ClientCertificates.Add(clientCert);
+                httpClientHandler.ClientCertificates.Add(clientCert);
                 _usingX509ClientCert = true;
             }
 
             IWebProxy proxy = iotHubClientHttpSettings.Proxy;
             if (proxy != null)
             {
-                _httpClientHandler.UseProxy = true;
-                _httpClientHandler.Proxy = proxy;
+                httpClientHandler.UseProxy = true;
+                httpClientHandler.Proxy = proxy;
             }
 
-            _httpClientHandler.MaxConnectionsPerServer = DefaultMaxConnectionsPerServer;
+            httpClientHandler.MaxConnectionsPerServer = DefaultMaxConnectionsPerServer;
             ServicePoint servicePoint = ServicePointManager.FindServicePoint(_baseAddress);
             servicePoint.ConnectionLeaseTimeout = s_defaultConnectionLeaseTimeout.Milliseconds;
 
-            _httpClientObj = new HttpClient(_httpClientHandler)
+            _httpMessageHandler = httpClientHandler;
```

Comment:
why not do this initialization on L64 directly?

Reply:
```_httpMessageHandler``` is the interface and only the implementation classes have fields for proxy, connection lease timeout, etc.

Reply:
Actually, scratch that. That was the case in v1, but here in v2 I can simplify this